### PR TITLE
Add `slowest_modules` option to `mix test`

### PR DESCRIPTION
**Pull request by gpopides**
Based on Jose's comment in this PR https://github.com/elixir-lang/elixir/pull/13283 he mentions that an option that logs the slowest modules during testing would be helpful

This PR addresses that comment.

I followed the same approach with `slowest` option implementation.

### CODE REVIEW

File: lib/ex_unit/test/ex_unit_test.exs [hunk 1]
```diff
@@ -239,6 +239,42 @@ defmodule ExUnitTest do
...
+        Process.sleep(12)
+        refute false
+      end
+    end
+
+    defmodule SlowestTestModule do
+      use ExUnit.Case
+
+      test "slowest" do
+        Process.sleep(15)
```
Comment (josevalim) [inline lib/ex_unit/test/ex_unit_test.exs:265]:
```suggestion
        refute false
      end
    end

    defmodule SlowerTestModule do
      use ExUnit.Case

      test "slower" do
        Process.sleep(5)
        refute false
      end
    end

    defmodule SlowestTestModule do
      use ExUnit.Case

      test "slowest" do
        Process.sleep(10)
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

File: lib/ex_unit/lib/ex_unit/cli_formatter.ex [hunk 2]
```diff
@@ -16,6 +16,7 @@ defmodule ExUnit.CLIFormatter do
       colors: colors(opts),
       width: get_terminal_width(),
       slowest: opts[:slowest],
+      slowest_modules: opts[:slowest_modules],
```
Comment (wojtekmach) [inline lib/ex_unit/lib/ex_unit/cli_formatter.ex:19]: was name :slowest_cases considered? I thought case is a more common used name for this even though it is of cluese modules under the hood.

Comment (gpopides) [reply lib/ex_unit/lib/ex_unit/cli_formatter.ex:19]: you have a point, i didn't think of it. My only thought about this is that a single test is also a case, isn't it?

Comment (josevalim) [reply lib/ex_unit/lib/ex_unit/cli_formatter.ex:19]: Yes, we moved away from the case naming because it was ambiguous. Today we have Test and TestModule.

**Pull request merged**